crates/client-api/src/routes/database.rs

```diff
@@ -885,10 +866,8 @@ pub struct DeleteDatabaseParams {
 pub async fn delete_database<S: ControlStateDelegate>(
     State(ctx): State<S>,
     Path(DeleteDatabaseParams { address }): Path<DeleteDatabaseParams>,
-    auth: SpacetimeAuthHeader,
+    Extension(auth): Extension<SpacetimeAuth>,
 ) -> axum::response::Result<impl IntoResponse> {
-    let auth = auth_or_unauth(auth)?;
```

Comment:
(review note: I think this used to fail if auth wasn't present, but it's fine because `delete_database` checks the identity)